warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/partition/SortedTabletLocationPartitioner.java

```diff
@@ -0,0 +1,80 @@
+
+import com.google.common.collect.TreeMultimap;
+import datawave.ingest.mapreduce.job.SplitsFileType;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+
+public class SortedTabletLocationPartitioner extends MultiTableRangePartitioner {
+    
+    private static final Logger log = Logger.getLogger(SortedTabletLocationPartitioner.class);
+    private final Map<String,Map<Integer,Integer>> SPLIT_TO_REDUCER_MAP = new HashMap<>();
+    
+    @Override
+    protected int calculateIndex(int index, int numPartitions, String tableName, int cutPointArrayLength) {
+        if (isAssignedPartition(tableName, index)) {
+            return SPLIT_TO_REDUCER_MAP.get(tableName).get(index);
+        }
+        assignPartitions(numPartitions, tableName, cutPointArrayLength);
+        return isAssignedPartition(tableName, index) ? SPLIT_TO_REDUCER_MAP.get(tableName).get(index) : 0;
+        
+    }
+    
+    private void assignPartitions(int numPartitions, String tableName, int cutPointArrayLength) {
+        Map<Text,String> currentTableSplitToLocation = splitToLocationMap.get().get(tableName);
+        Map<Integer,Integer> tempSplitReducerMap = new HashMap<>();
+        Text[] cutPointArray = splitsByTable.get().get(tableName);
+        
+        if (cutPointArrayLength > numPartitions) {
+            mapPartitions(numPartitions, cutPointArrayLength, currentTableSplitToLocation, tempSplitReducerMap, cutPointArray);
+        } else {
+            for (int i = 0; i < cutPointArrayLength; i++) {
+                tempSplitReducerMap.put(i, i);
+                tempSplitReducerMap.put(-i - 1, i);
+            }
+        }
+        
+        SPLIT_TO_REDUCER_MAP.put(tableName, tempSplitReducerMap);
+    }
+    
+    private void mapPartitions(int numPartitions, int cutPointArrayLength, Map<Text,String> currentTableSplitToLocation,
+                    Map<Integer,Integer> tempSplitReducerMap, Text[] cutPointArray) {
+        
+        int locationsAssigned = 0;
+        int assignedReducer = 0;
+        
+        TreeMultimap<String,Integer> locationToSplits = TreeMultimap.create();
+        
+        for (int k = 0; k < cutPointArrayLength; k++) {
+            locationToSplits.put(currentTableSplitToLocation.get(cutPointArray[k]), k);
+        }
+        
+        Iterator<String> locationIterator = locationToSplits.keySet().iterator();
+        while (locationIterator.hasNext()) {
+            Set<Integer> splitsForCurrentLocation = locationToSplits.get(locationIterator.next());
+            
+            for (Integer splitIndex : splitsForCurrentLocation) {
+                tempSplitReducerMap.put(splitIndex, assignedReducer);
+                tempSplitReducerMap.put(-splitIndex - 1, assignedReducer);
+            }
+            
+            locationsAssigned++;
+            // simple round robin for now. if accumulo isn't balanced we will feel the effects here and will likely want to add constraints at some point
```

Comment:
agreed.  Perhaps we should keep track of the number of splits assigned to a reducer and assign the next location to to a reducer with the least number of assigned splits.  This could be done simply with a heap.